finetune/target_models/sequence_labeling.py

```diff
@@ -85,7 +86,47 @@ def _target_encoder(self):
             return SequenceMultiLabelingEncoder(pad_token=self.config.pad_token)
         return SequenceLabelingEncoder(pad_token=self.config.pad_token)
 
-
+def _context_same_cluster(a, b, thresholds=None):
+    thresholds = thresholds or {"lr": 0.5, "tb": 0.5}
+    if not isinstance(thresholds, dict):
+        thresholds = {"lr": thresholds, "tb": thresholds}
+
+    if "lr" not in thresholds or "tb" not in thresholds:
+        raise FinetuneError("threshold dict {} does not contain the required keys lr, and tb")
+        
+    for ai in a:
+        atb = abs(ai["bottom"] - ai["top"])
+        alr = abs(ai["right"] - ai["left"])
+        for bi in b:
+            btb = abs(bi["bottom"] - bi["top"])
+            blr = abs(bi["right"] - bi["left"])
+            
+            ttb = max(btb, atb)
+            tlr = max(blr, alr)
+
+            if min(abs(ai["left"] - bi["right"]), abs(ai["right"] - bi["left"])) < thresholds["lr"] * tlr:
+                return True
+            if min(abs(ai["top"] - bi["bottom"]), abs(ai["bottom"] - bi["top"])) < thresholds["tb"] * ttb:
```

Comment:
I think we should only be checking one of these dimensions, i.e. the one opposite to the dimension where we detected the overlap